### PR TITLE
Adding telemetry directory cleanup

### PR DESCRIPTION
**Pull request by peschkaj**
This should implement #340

### CODE REVIEW

File: src/rustup/telemetry.rs [hunk 1]
```diff
@@ -47,6 +49,43 @@ impl Telemetry {
...
+        // If it is a new day, then attempt to clean the telemetry directory.
+        if !raw::is_file(&self.telemetry_dir.join(&filename)) {
+            self.clean_telemetry_dir();
+        }
+
         let _ = utils::append_file("telemetry", &self.telemetry_dir.join(&filename), &json);
     }
+
+    pub fn clean_telemetry_dir(&self) {
+        let contents = self.telemetry_dir.read_dir().unwrap();
```
Comment (brson) [inline src/rustup/telemetry.rs]: Please handle this error instead of unwrapping.

File: src/rustup/telemetry.rs [hunk 2]
```diff
@@ -47,6 +49,43 @@ impl Telemetry {
...
+
+        for c in contents {
+            let x = c.unwrap();
+            let filename = x.path().file_name().unwrap().to_str().unwrap().to_owned();
+            if filename.starts_with("log") && filename.ends_with("json") {
+                telemetry_files.push(x.path());
+            }
+        }
+
+        println!("found {:?} files to delete", telemetry_files.len());
```
Comment (brson) [inline src/rustup/telemetry.rs]: Let's do this silently and not print anything.

File: src/rustup/telemetry.rs [hunk 3]
```diff
@@ -47,6 +49,43 @@ impl Telemetry {
...
+
+        let dl: usize = telemetry_files.len() - MAX_TELEMETRY_FILES;
+        let dl = dl + 1 as usize;
+
+        telemetry_files.sort();
+        telemetry_files.dedup();
+
+        for i in 0..dl {
+            let i = i as usize;
+            let _ = fs::remove_file(&telemetry_files[i]);
```
Comment (brson) [inline src/rustup/telemetry.rs]: Please handle this error too.

**Comment by brson** [comment]
Sweet. Looks good but for a few comments. Let me know if you have difficulties with the error handling. It can be awkward in rustup because the error types don't have automatic `From` conversions.

**Comment by peschkaj** [comment]
How far up the stack should these errors go?  I can change `clean_telemetry_dir` to return a `Result<()>`. This seems like something we might want to do in the event that we can't execute `read_dir()` on the directory or that we can't delete a file from the telemetry directory. 

I was thinking that an error in `clean_telemetry_dir` should be caught in `log_telemetry` and then if an error occurs, I should use one of the `info!` macro to let the user know that we couldn't delete old log files.

**Comment by brson** [comment]
@peschkaj Agh. Sorry for the delay in responding again. You can always ping me on IRC if I don't respond quickly.

I'd prefer to just bubble them all the way up to the top by returning `Result`s. That's how all other I/O errors are dealt with presently.

**Comment by brson** [comment]
Thanks again!

**Pull request merged**